### PR TITLE
created role entity

### PR DESCRIPTION
**Pull request by thebadpanda**

### CODE REVIEW

File: src/main/java/com/softserve/actent/model/entity/Role.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.softserve.actent.model.entity;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.hibernate.validator.constraints.Length;
+
+import javax.persistence.*;
```
Comment (md1guy) [inline src/main/java/com/softserve/actent/model/entity/Role.java]: Replace with single particular imports.

File: src/main/java/com/softserve/actent/model/entity/Role.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import javax.validation.constraints.NotBlank;
+
+@Entity
+@Table(name = "roles")
+@Data
+@NoArgsConstructor
```
Comment (md1guy) [inline src/main/java/com/softserve/actent/model/entity/Role.java:18]: Change annotations order: hibernate annotations must be the last.

File: src/main/java/com/softserve/actent/model/entity/Role.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@Entity
+@Table(name = "roles")
+public class Role {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "role", unique = true)
```
Comment (Senchakmen) [inline src/main/java/com/softserve/actent/model/entity/Role.java]: The name of the field is the same with Column name

**Review by octap16 — DISMISSED** [review]

**Review by Senchakmen — APPROVED** [review]

File: src/main/java/com/softserve/actent/model/entity/Role.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+@NoArgsConstructor
+@Entity
+@Table(name = "roles")
+public class Role {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+
```
Comment (octap16) [inline src/main/java/com/softserve/actent/model/entity/Role.java:27]: delete empty field

**Review by nazarmavko — APPROVED** [review]

**Review by onlyVone-zz — APPROVED** [review]

File: src/main/java/com/softserve/actent/model/entity/Role.java (hunk 4 above)
Comment (md1guy) [reply src/main/java/com/softserve/actent/model/entity/Role.java:27]: excessive string

**Review by Senchakmen — APPROVED** [review]

File: src/main/java/com/softserve/actent/model/entity/Role.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    @NonNull
+    @Length(max = 20,  message = "Too long")
+    @NotBlank(message = "Role cannot be empty")
+    @Column(unique = true, nullable = false)
+    private String role;
```
Comment (Pitcity) [inline src/main/java/com/softserve/actent/model/entity/Role.java:32]: Role is string or enum?

Check how enum stores in DB

**Pull request merged**